define/types.go

```diff
@@ -186,16 +191,52 @@ func cloneToDirectory(url, dir string) ([]byte, string, error) {
 		}
 		gitBranch = gitSubDirPart[0]
 	}
-	if gitBranch == "" {
-		logrus.Debugf("cloning %q to %q", gitBranchPart[0], dir)
-		cmd = exec.Command("git", "clone", "--recurse-submodules", gitBranchPart[0], dir)
-	} else {
-		logrus.Debugf("cloning repo %q and branch %q to %q", gitBranchPart[0], gitBranch, dir)
-		cmd = exec.Command("git", "clone", "--recurse-submodules", "-b", gitBranch, gitBranchPart[0], dir)
-	}
+	return gitBranchPart[0], gitSubdir, gitBranch
+}
 
+func cloneToDirectory(url, dir string) ([]byte, string, error) {
+	var cmd *exec.Cmd
+	gitRepo, gitSubdir, gitBranch := parseGitBuildContext(url)
+	// init repo
+	cmd = exec.Command("git", "init", dir)
 	combinedOutput, err := cmd.CombinedOutput()
-	return combinedOutput, gitSubdir, err
+	if err != nil {
+		return combinedOutput, gitSubdir, fmt.Errorf("failed while performing `git init`: %w", err)
+	}
+	// add origin
+	cmd = exec.Command("git", "remote", "add", "origin", gitRepo)
+	cmd.Dir = dir
+	combinedOutput, err = cmd.CombinedOutput()
+	if err != nil {
+		return combinedOutput, gitSubdir, fmt.Errorf("failed while performing `git remote add`: %w", err)
+	}
+	// fetch required branch or commit and perform checkout
+	// Always default to `HEAD` if nothing specified
+	fetch := "HEAD"
+	if gitBranch != "" {
+		fetch = gitBranch
+	}
+	logrus.Debugf("fetching repo %q and branch (or commit ID) %q to %q", gitRepo, gitBranch, dir)
```

Comment:
```suggestion
	logrus.Debugf("fetching repo %q and branch (or commit ID) %q to %q", gitRepo, fetch, dir)
```
`gitBranch` could be empty, and it's not what we're passing to `git`.